server/db/dbManager.ts

```diff
@@ -35,7 +35,7 @@ export class PostManager {
 		return this.model.countDocuments().exec();
 	}
 
-	async getPostsPage(pageSize: number | string, pageNum: number | string): Promise<models.IPost[]> {
+	async getPostsPage(pageSize: number | string, pageNum: number | string, sort: 'default' | 'recent' | 'most-upvotes' | 'oldest'): Promise<models.IPost[]> {
```

Comment:
Use the enum here and just set the default value.
```suggestion
         enum SortOption {
             DEFAULT,
             RECENT,
             ....
         }
	async getPostsPage(pageSize: number | string, pageNum: number | string, sort: SortOption = SortOption.DEFAULT): Promise<models.IPost[]> {
```